File: js/graph.js
function Edge(n1,n2) {
  this.n1 = n1;
  this.n2 = n2;
  this.distance = dist(n1,n2);
  this.same = (n1.category == n2.category);

  this.weight = function() {
    p_mov0 = 0.5;
    dropoff = 0.01;
    w = p_mov0-p_mov0/(1+Math.exp(-dropoff*this.distance));
    if (self.same) {
      return w*1.2;
    }
    else {
      return w;
    }
  }
}

function Venue(json, id) {
  this.id = id
  this.name = json['name'];
  this.category = json['category'];
  this.lat = json['lat'];
  this.lng = json['lng'];
  this.population = json['population'];
  this.infected = 0;
  this.edges = [];
  this.contentString = function() {
    return (this.name + "\n Hipsters: " + this.infected)
  }
}

function distance_on_sphere(lat1, lon1, lat2, lon2) {
  var radlat1 = Math.PI * lat1/180
    var radlat2 = Math.PI * lat2/180
    var radlon1 = Math.PI * lon1/180
    var radlon2 = Math.PI * lon2/180
    var theta = lon1-lon2
    var radtheta = Math.PI * theta/180
    var dist = Math.sin(radlat1) * Math.sin(radlat2) + Math.cos(radlat1) * Math.cos(radlat2) * Math.cos(radtheta);
    dist = Math.acos(dist)
    dist = dist * 180/Math.PI
    dist = dist * 60 * 1.1515
    dist = dist * 1609.344 //Get km
    return dist
}
function dist(n1, n2) {
  return distance_on_sphere(n1.lat,n1.lng,n2.lat,n2.lng)
}

File: old/graph.py
import math
class Edge:
  def __init__(self, n1,n2):
    self.n1 = n1
    self.n2 = n2
    self.distance = dist(n1,n2)
    self.same = (n1.category == n2.category)
    self.weight = calc_weight(self.dist)

    def calc_weight(self,dist):
      w = p_mov0-p_mov0/(1+exp(-dropoff*dist))
      if(same):
        return w*1.2
      return w

class Venue:
  def __init__(self, json, category_name):
    self.json = json #store the whole object just in case
    self.name = json['name']
    self.category = category_name
    self.lat = int(json['location']['lat'])
    self.lng = int(json['location']['lng'])
    self.pop = int(json['stats']['usersCount'])
    self.infected = 0
    self.edges = []

def distance_on_sphere(lat1, long1, lat2, long2):
    degrees_to_radians = math.pi/180.0
    phi1 = (90.0 - lat1)*degrees_to_radians
    phi2 = (90.0 - lat2)*degrees_to_radians
    theta1 = long1*degrees_to_radians
    theta2 = long2*degrees_to_radians
    cos = (math.sin(phi1)*math.sin(phi2)*math.cos(theta1 - theta2) + math.cos(phi1)*math.cos(phi2))
    arc = math.acos( cos )
    return arc*6378100.0

def dist(n1, n2):
  distance_on_sphere(n1.lat,n1.lng,n2.lat,n2.lng)
